### ISSUE TITLE
create_function() is deprecated in PHP 7.2

### ISSUE THREAD

**Issue by kevinlisota** [Automattic/batcache#83]
`create_function` should be replaced with an anonymous function for PHP 7.2 support. `create_function` is deprecated in PHP 7.2

**Comment by TomasVotruba** [comment]
Hyi, if you need to upgrade more than one `create_function` to anonymous function ([official recommended upgrade](http://php.net/manual/en/migration72.deprecated.php#migration72.deprecated.create_function-function)), you can use a CLI tool for that.

It's tested on [30 various (and really weird :)) cases](https://github.com/rectorphp/rector/tree/c94f3208c0275c66385931e497afaf1001dfcc1e/packages/Php/tests/Rector/FuncCall/CreateFunctionToAnonymousFunctionRector/Fixture).

```diff
-$callback = create_function('$a', 'return "<cas:proxy>$a</cas:proxy>";');    
+$callback = function ($a) {
+    return "<cas:proxy>{$a}</cas:proxy>";
+};
```

Includes concat (.), string quotes and inclined function calls:

```diff
-$func = create_function('$atts, $content = null','return "<div class=\"' . $class_list . '\">" . do_shortcode($content) . "</div>";' );
+$func = function ($atts, $content = null) use ($class_list) {
+    return "<div class=\"{$class_list}\">" . do_shortcode($content) . "</div>";
+};
```

Do you want to automate the hard work?


### 1. Instal Rector

```bash
composer require rector/rector --dev
```

### 2. Create config

```yaml
# rector.yml
services:
    Rector\Php\Rector\FuncCall\CreateFunctionToAnonymousFunctionRector: ~
```

### 3. Upgrade your Code


```bash
vendor/bin/rector process src --config rector.yml --dry-run
vendor/bin/rector process src --config rector.yml
```

**Comment by jeherve** [comment]
This should be fixed as of #103.
